src/lib/crypto/s2k.cpp

```diff
@@ -157,6 +157,10 @@ pgp_s2k_compute_iters(pgp_hash_alg_t alg, size_t desired_msec, size_t trial_msec
     const uint8_t MIN_ITERS = 96;
     uint8_t       buf[8192] = {0};
     size_t        bytes = 0;
+    pgp_hash_t    hash = {};
+    uint64_t      start;
+    uint64_t      end;
+    uint64_t      duration;
```

Comment:
We don't really have anything official in our C++ guide on this, but I think it's standard practice to make variable declarations as late as possible, so I don't like seeing all the variables moved up top like this (and they should all probably be initialized).

Refs:
* C++ Coding Standards: 101 Rules, Guidelines, and Best Practices
`Declare variables as locally as possible.`
* [Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html#Local_Variables)
`Place a function's variables in the narrowest scope possible, and initialize variables in the declaration.`
* [Mozilla Coding Style](https://developer.mozilla.org/en-US/docs/Mozilla/Developer_guide/Coding_Style#General_C.2FC.2B.2B_Practices)
`Declare local variables as near to their use as possible.`

Reply:
Thanks, @dewyatt ! Honestly I believed (without any reference :) ) that declaration of variables on very top of the function is best practice, but that was wrong. At least for the C++.
Will fix and update PR soon.